youtube_dl/extractor/plutotv.py

```diff
@@ -0,0 +1,162 @@
+                'season_number': 2,
+                'episode_number': 3,
+                'duration': 3600,
+            }
+        },
+        {
+            'url': 'https://pluto.tv/on-demand/series/i-love-money/season/1/',
+            'playlist_count': 11,
+            'info_dict': {
+                'id': '5de6c582e9379ae4912dedbd',
+                'title': 'I Love Money - Season 1',
+            }
+        },
+        {
+            'url': 'https://pluto.tv/on-demand/series/i-love-money/',
+            'playlist_count': 26,
+            'info_dict': {
+                'id': '5de6c582e9379ae4912dedbd',
+                'title': 'I Love Money',
+            }
+        },
+        {
+            'url': 'https://pluto.tv/on-demand/movies/arrival-2015-1-1',
+            'md5': '3cead001d317a018bf856a896dee1762',
+            'info_dict': {
+                'id': '5e83ac701fa6a9001bb9df24',
+                'ext': 'mp4',
+                'title': 'Arrival',
+                'description': 'When mysterious spacecraft touch down across the globe, an elite team - led by expert translator Louise Banks (Academy Award® nominee Amy Adams) – races against time to decipher their intent.',
+                'duration': 9000,
+            }
+        },
+    ]
+
+    def _to_ad_free_formats(self, video_id, formats):
+        ad_free_formats = []
+        for format in formats:
+            res = self._download_webpage_handle(
+                format.get('url'), video_id, note='Downloading m3u8 playlist',
+                fatal=False)
+            if res is False:
+                continue
+            playlist_doc, _ = res
+            first_segment_url = re.search(
+                r'^(https?://.*/)0\-(end|[0-9]+)/[^/]+\.ts$', playlist_doc,
+                re.MULTILINE)
+            if not first_segment_url:
+                continue
+            m3u8_url = compat_urlparse.urljoin(
+                first_segment_url.group(1), '0-end/master.m3u8')
+            ad_free_formats.extend(
+                self._extract_m3u8_formats(
+                    m3u8_url, video_id, 'mp4', 'm3u8_native',
+                    m3u8_id='hls', fatal=False))
+        return ad_free_formats
+
+    def _get_video_info(self, video_json, slug, series_name=None):
+        video_id = video_json.get('_id', slug)
+        formats = []
+        for video_url in video_json.get('stitched', {}).get('urls', []):
+            if video_url.get('type') != 'hls':
+                continue
+            url = video_url.get('url')
+            formats.extend(
+                self._extract_m3u8_formats(
+                    url, video_id, 'mp4', 'm3u8_native',
+                    m3u8_id='hls', fatal=False))
+        self._sort_formats(formats)
+        ad_free_formats = self._to_ad_free_formats(video_id, formats)
+        self._sort_formats(ad_free_formats)
+        info = {
+            'id': video_id,
+            'formats': ad_free_formats if ad_free_formats else formats,
```

Comment:
`ad_free_formats` is never empty here since `_sort_formats` will throw if's empty.